src/components/cylc/tree/deltas.js

```diff
@@ -181,71 +176,76 @@ function handleInitialDataBurst (deltas, tree) {
  * the first family from the top of the hierarchy in the deltas.
  *l
  * @param {Deltas} deltas - GraphQL deltas
- * @param {CylcTree} tree - Tree object backed by an array and a Map
+ * @param {Workflow} workflow - Tree object
+ * @param {Lookup} lookup
+ * @param {*} options
  */
-function handleDeltas (deltas, tree) {
-  if (deltas.pruned) {
-    applyDeltasPruned(deltas.pruned, tree)
-  }
-  if (deltas.added) {
-    applyDeltasAdded(deltas.added, tree)
-  }
-  if (deltas.updated) {
-    applyDeltasUpdated(deltas.updated, tree)
-  }
+function handleDeltas (deltas, workflow, lookup, options) {
+  const errors = []
+  Object.keys(DELTAS).forEach(key => {
+    if (deltas[key]) {
+      const handlingFunction = DELTAS[key]
+      const result = handlingFunction(deltas.added, workflow, lookup, options)
+      errors.push(...result.errors)
+    }
+  })
   // if added, removed, or updated deltas, we want to re-calculate the cycle point states now
   if (deltas.pruned || deltas.added || deltas.updated) {
-    tree.tallyCyclePointStates()
+    // FIXME
```

Comment:
Note to self: remove this comment. I had commented the `tallyCyclePointStates()` call, and marked with a NOTE (the IDE highlights it, so it's easier for me to see what's pending), but forgot to remove the mark.